pallets/delegation/src/delegation_hierarchy.rs

```diff
@@ -69,17 +71,28 @@ pub struct DelegationNode<T: Config> {
 	pub children: BoundedBTreeSet<DelegationNodeIdOf<T>, T::MaxChildren>,
 	/// The additional information attached to the delegation node.
 	pub details: DelegationDetails<T>,
+	/// The deposit that was taken to incentivise fair use of the on chain
+	/// storage.
+	pub deposit: Deposit<AccountIdOf<T>, BalanceOf<T>>,
 }
 
 impl<T: Config> DelegationNode<T> {
 	/// Creates a new delegation root node with the given ID and delegation
 	/// details.
-	pub fn new_root_node(id: DelegationNodeIdOf<T>, details: DelegationDetails<T>) -> Self {
+	pub fn new_root_node(
+		id: DelegationNodeIdOf<T>,
+		details: DelegationDetails<T>,
+		deposit_owner: AccountIdOf<T>,
+	) -> Self {
 		Self {
 			hierarchy_root_id: id,
 			parent: None,
 			children: BoundedBTreeSet::<DelegationNodeIdOf<T>, T::MaxChildren>::new(),
 			details,
+			deposit: Deposit::<AccountIdOf<T>, BalanceOf<T>> {
+				owner: deposit_owner,
+				amount: <T as Config>::Deposit::get(),
```

Comment:
M(reserve-deposit part 4): if we hard code the deposit here, we never made sure that the deposit was reserved somewhere. So we might end up to store a deposit without reserving.